### DISCUSSION TITLE
Send signed transaction in JSON format through RPC meets error.

### DISCUSSION THREAD

**Discussion by OstNo1** [aptos-labs/aptos-developer-discussions#394] [Questions]
### Describe your question in detail.

I send a request of the following JSON body to bocast a transaction through RPC : https://api.mainnet.aptoslabs.com/v1/transactions POST, but got an error:  **Failed to parse transaction payload: parse arguments[0] failed, expect string<move_struct_tag_id>, caused by error: invalid type: map, expected a string**. I think this JSON arguments[0] is correct according to the address 0x4bf51972879e3b95c4781a5cdcb9e1ee24ef483e7d22f2d903626f126df62bd1 ABI on chain, please help to check what is wrong with it, thx.
`{
    "sender": "my address",
    "sequence_number": "84",
    "max_gas_amount": "100000",
    "gas_unit_price": "100",
    "expiration_timestamp_secs": "1725370236",
    "payload": {
        "type": "entry_function_payload",
        "function": "0x4bf51972879e3b95c4781a5cdcb9e1ee24ef483e7d22f2d903626f126df62bd1::router::add_liquidity_and_stake_coin_entry",
        "type_arguments": [
            "0x1::aptos_coin::AptosCoin"
        ],
        "arguments": [
            {
                "inner": "0x2ebb2ccac5e027a87fa0e2e5f656a3a4238d6a48d93ec9b610d570fc0aa0df12"
            },
            false,
            "1308296",
            "199999885"
        ]
    },
    "signature": {
        "type": "ed25519_signature",
        "public_key": "my public key",
        "signature": "my signature"
    }
}`

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

REST API

**Comment by 0xmaayan** [comment]
The arguments property should be an array of strings and not a json. Try sending the address as a string

arguments:["0x123...789"]

**Reply by OstNo1** [reply]
Thanks for reaching out. But I wonder why this is a String not a map {"inner": "xxxx"}, I got other successful transaction request data(the follow data), the main difference is the arg type according to the contract on chain:  failed one is : 0x1::object::Object<0x1::fungible_asset::Metadata> ,  success one is :  vector<0x1::object::Object<0x1::fungible_asset::Metadata>>

successful transaction data:
`{
    "sender": "my address",
    "sequence_number": "79",
    "max_gas_amount": "100000",
    "gas_unit_price": "100",
    "expiration_timestamp_secs": "1725359226",
    "payload": {
        "type": "entry_function_payload",
        "function": "0x4bf51972879e3b95c4781a5cdcb9e1ee24ef483e7d22f2d903626f126df62bd1::router::swap_route_entry_from_coin",
        "type_arguments": [
            "0x1::aptos_coin::AptosCoin"
        ],
        "arguments": [
            "10000000",
            "1490574843",
            [
                {
                    "inner": "0xbeccb00e5082bc3ef9f20e08d46edab77f7b1c7bfc8a21d066a4269abd825843"
                },
                {
                    "inner": "0x2ebb2ccac5e027a87fa0e2e5f656a3a4238d6a48d93ec9b610d570fc0aa0df12"
                }
            ],
            [
                false,
                false
            ],
            "my address"
        ]
    },
    "signature": {
        "type": "ed25519_signature",
        "public_key": "my public key",
        "signature": "my signature"
    }
}`
